fluent-bundle/tests/bundle.rs

```diff
@@ -19,19 +18,75 @@ fn add_resource_override() {
 
     let value = bundle
         .get_message("key")
-        .expect("Failed to retireve a message")
+        .expect("Failed to retrieve a message")
         .value()
-        .expect("Failed to retireve a value of a message");
+        .expect("Failed to retrieve a value of a message");
     assert_eq!(bundle.format_pattern(value, None, &mut errors), "Value");
 
     bundle.add_resource_overriding(&res2);
 
     let value = bundle
         .get_message("key")
-        .expect("Failed to retireve a message")
+        .expect("Failed to retrieve a message")
         .value()
-        .expect("Failed to retireve a value of a message");
+        .expect("Failed to retrieve a value of a message");
     assert_eq!(bundle.format_pattern(value, None, &mut errors), "Value 2");
 
     assert!(errors.is_empty());
 }
+
+#[test]
+fn borrowed_plain_message() {
+    let res = FluentResource::try_new("key = Value".to_string()).unwrap();
+    let en_us = langid!("en-US");
+
+    let mut bundle = FluentBundle::new(vec![en_us]);
+    bundle.add_resource(&res).expect("Failed to add a resource");
+
+    let mut errors = vec![];
+
+    let formatted_pattern = {
+        let value = bundle
+            .get_message("key")
+            .expect("Failed to retrieve a message")
+            .value()
+            .expect("Failed to retrieve a value of a message");
+
+        bundle.format_pattern(value, None, &mut errors)
+    };
+
+    fn is_borrowed(cow: Cow<'_, str>) -> bool {
+        match cow {
+            Cow::Borrowed(_) => true,
+            _ => false,
+        }
+    }
+    assert_eq!(formatted_pattern, "Value");
+    assert!(is_borrowed(formatted_pattern));
+}
+
+#[test]
+fn arguments_outlive_formatted_pattern() {
+    let res = FluentResource::try_new("key = { $variable }".to_string()).unwrap();
+    let en_us = langid!("en-US");
+
+    let mut bundle = FluentBundle::new(vec![en_us]);
+    bundle.add_resource(&res).expect("Failed to add a resource");
+
+    let mut errors = vec![];
+
+    let formatted_pattern = {
```

Comment:
Thought: Yes this makes for a very convincing example of why this change is needed.